### PR TITLE
11461 register torque layer dependencies

### LINKED ISSUE #11461: Torque layer dependencies not correctly registered
### Context
Due to a confusion with the meaning of `Layer.data_layer` and `Map.data_layers` (they are not the same), there are a bunch of pieces of code that are incorrect. This ticket illustrates one of them. 

### Steps to Reproduce
1. Create a map
2. Add a torque layer on dataset P
3. Add a join analysis to the torque layer, with dataset Q
4. Try to delete dataset Q

### Current Result
You are not warned of a dependency with the map .

### Expected result
You should be warned.

### Additional info
The problem is that `Carto::Map.update_dataset_dependencies` calls `data_layers` which does not include torque layers. This mistake is repeated throughout the code (I'm thinking that the layer node style cache might also be affected).

My suggestion here would be to bite the bullet, and instead of fixing this specific issue (it's as easy as changing `data_layers` to `carto_and_torque_layers`), changing the behaviour of `data_layers` so it behaves consistently with `Layer.data_layer?`.

### PR DESCRIPTION
**Pull request by javitonino**
Fixes #11461

Makes the following important changes:
- Adds `Map.carto_layers` to return layers with kind carto
- `Map.data_layers` now returns carto and torque layers
- Removed `Map.carto_and_torque_layers`

This makes the interface at map level more similar to the one at layer level. There is an overview table at the ticket. There are more changes that could be made (`Map.base_layers` is a good candidate, since it returns all layers), but this only focuses in the one producing the actual issue.

This fixes the following:
- No update table dependencies on torque layers
- No auto-indexing on torque layers (when publishing a map)
- No layer node style on torque layers presenter (little effect, since the frontend switches torque layers to normal after applying an analysis)
- No updated_at date in torque layers
- Incorrect calculation of max layers per map (not counting torque layers)

**Acceptance** (suggestions, this one is hard to evaluate)
- [x] STR of the ticket
- [x] Create an animated torque layer over a dataset with > 10k rows. A timeline widget should be automatically added. Check Rollbar, you should see an autoindex trigger.
- ~~In a map with a Torque layer, try adding a new Torque layer (APII/console, we don't want to test frontend blocking). You shouldn't be able to.~~ Not needed. There's no public API for this.
- [x] Export a .carto from a map with labels on top, a normal layer and a torque layer. Import it. It should keep layers and order.
- [x] Repeat with a map using a basemap without labels.

### CODE REVIEW

File: app/controllers/carto/api/visualization_vizjson_adapter.rb [hunk 1]
```diff
@@ -8,7 +8,7 @@ class VisualizationVizJSONAdapter
       include Carto::HtmlSafe
```
Comment (javitonino) [inline app/controllers/carto/api/visualization_vizjson_adapter.rb]: Self comment. Add NEWS entry.

File: app/controllers/carto/api/vizjson3_presenter.rb [hunk 2]
```diff
@@ -124,7 +124,7 @@ def bounds_from(map)
       def layers_vizjson(forced_privacy_version)
         ordered_layers = (
           [basemap_layer] +
-          @visualization.data_layers +
+          @visualization.carto_layers +
```
Comment (juanignaciosl) [inline app/controllers/carto/api/vizjson3_presenter.rb]: Doesn't this exclude torque layers?

(visualization.carto_layers = map.carto_layers = layers.kind == carto)

Shouldn't this be `data_layers`? It looks like it was previously excluded as well, but I'm not sure if this should be fixed.

**Comment by juanignaciosl** [comment]
More acceptance:
- [ ] In a map with a Torque layer, try adding a new Torque layer. You shouldn't be able to.
- [ ] Export a .carto from a map with labels on top, a normal layer and a torque layer. Import it. It should keep layers and order.
- [ ] Repeat with a map using a basemap without layers.

File: app/controllers/carto/api/vizjson3_presenter.rb (hunk 2 above)
Comment (javitonino) [inline app/controllers/carto/api/vizjson3_presenter.rb]: Yes, it does, that's the point. This is the entire code chunk, before the changes:

```
ordered_layers = (
  [basemap_layer] +
  @visualization.data_layers +
  @visualization.other_layers +
  non_basemap_base_layers).compact
```

It is building the list of layers in the order they should appear in the vizjson. So, we must keep `carto_layers` and `torque_layers` as separate calls and in that order, so the torque layer is always on top of the carto layers in the vizjson.

I tried to replace `carto_layers + torque_layers` for `data_layers`, but then I realized (tests are great!) that this was a bad idea.

Comment (juanignaciosl) [inline app/controllers/carto/api/vizjson3_presenter.rb]: Oh, I was misled by `other_layers`, which is based on `reject` 👍

**Comment by javitonino** [comment]
Added your acceptance suggestions to the top list. Ready to go?

**Pull request merged**